### ISSUE TITLE
Expected non-null name in imported array child 

### ISSUE THREAD

**Issue by JkSelf** [facebookincubator/velox#7690]
### Bug description

The `ConstVector `cannot be converted to the `ArrowSchema`. This results in an exception being thrown, specifically `Expected non-null name in imported array child`. The exception occurs because the name of the value schema is not set [here](https://github.com/facebookincubator/velox/blob/461434f34c292fe757c3fd05a6b90d55c13691ce/velox/vector/arrow/Bridge.cpp#L1035). Even when setting the name to 'values', the following exception still persists: `Operator::getOutput failed for [operator: TableWrite, plan node ID: 2]: NotImplemented: Unhandled type for Arrow to Parquet schema conversion: run_end_encoded<run_ends: int32, values: int64>`.

 ```
   ArrowSchema arrowSchema;
    const bool isScalar = (type->size() == 0);
    const bool constantSize = 100;

    // If scalar, create the constant vector directly; if complex type, create a
    // complex vector first, then wrap it in a dictionary.
    auto constantVector = isScalar
        ? BaseVector::createConstant(
              type, variant(type->kind()), constantSize, pool_.get())
        : BaseVector::wrapInConstant(
              constantSize,
              3, // index to use for the constant
              BaseVector::create(type, 100, pool_.get()));

    velox::exportToArrow(constantVector, arrowSchema);
    arrow::ImportSchema(&arrowSchema);
```

### System information

Velox System Info v0.0.2
Commit: 98d6c053c065cabe7e2071e040a9680efb4255f6
CMake Version: 3.16.3
System: Linux-5.4.0-156-generic
Arch: x86_64
C++ Compiler: /usr/bin/c++
C++ Compiler Version: 9.4.0
C Compiler: /usr/bin/cc
C Compiler Version: 9.4.0
CMake Prefix Path: /usr/local;/usr;/;/usr;/usr/local;/usr/X11R6;/usr/pkg;/opt

**Comment by JkSelf** [comment]
@mbasmanova @majetideepak Can you help to look this issue? Thanks.

**Comment by mbasmanova** [comment]
CC: @Yuhta @pedroerp @kgpai

**Comment by JkSelf** [comment]
@mbasmanova @majetideepak @Yuhta @pedroerp @kgpai Do you have any input? Thanks.

**Comment by majetideepak** [comment]
@JkSelf, @czentgr opened an issue here https://github.com/facebookincubator/velox/issues/7457.
The solution is to flatten the dictionary and constant vectors. PR https://github.com/facebookincubator/velox/pull/7025

**Comment by pedroerp** [comment]
@majetideepak I guess there is also a point to be made that if we don't flat constants while writing to Parquet, we could be smarter and leverage the constant encoding while writing - say we simply write it as a RLE to the Parquet file.

**Comment by majetideepak** [comment]
@pedroerp You are right. Now that we moved the Parquet writer code to Velox, we can implement writing Velox constant vector as Parquet RLE
https://parquet.apache.org/docs/file-format/data-pages/encodings/#a-namerlearun-length-encoding--bit-packing-hybrid-rle--3

**Comment by majetideepak** [comment]
The Parquet RLE piece is currently not implemented in Arrow as well. https://github.com/facebookincubator/velox/issues/7189#issuecomment-1800915223

**Comment by pedroerp** [comment]
#7802 fixes the Arrow conversion issue. Cc: @JkSelf

**Comment by JkSelf** [comment]
@pedroerp Thanks for your fix. With https://github.com/facebookincubator/velox/pull/7802 ,  `Expected non-null name in imported array child `exception no longer exists. However, the following exception occurs:
`Operator::getOutput failed for [operator: TableWrite, plan node ID: 2]: NotImplemented: Unhandled type for Arrow to Parquet schema conversion: run_end_encoded<run_ends: int32, values: int64>.`

**Comment by JkSelf** [comment]
@pedroerp Got it. We currently support the DataWriting operator in Gluten. However, we are awaiting resolution on this issue for static partition write. Are there any plans for upstream support of writing Constant/REE in the Parquet writer in the near future? Thanks.

**Comment by pedroerp** [comment]
@majetideepak and @czentgr are the ones working on Parquet reader/writer lately. Do you guys have any plans to look at this issue?

**Comment by JkSelf** [comment]
@majetideepak Thanks for your reply. https://github.com/facebookincubator/velox/pull/7025 can address my issues. Does converting the vector to a flattened form affect the performance or not?

**Comment by majetideepak** [comment]
> Does converting the vector to a flattened form affect the performance or not?

Unfortunately, the performance will degrade. We will prioritize implementing RLE Parquet.

**Comment by JkSelf** [comment]
@majetideepak Got it. Thanks for your reply.
